src/main/kotlin/org/neo4j/graphql/AugmentationHandler.kt

```diff
@@ -0,0 +1,22 @@
+package org.neo4j.graphql
+
+import graphql.schema.*
+
+abstract class AugmentationHandler(val schemaConfig: SchemaConfig) {
+    companion object {
+        const val QUERY = "Query"
+        const val MUTATION = "Mutation"
+    }
+
+    abstract fun augmentType(type: GraphQLFieldsContainer, buildingEnv: BuildingEnv)
+
+    abstract fun createDataFetcher(rootType: GraphQLObjectType, fieldDefinition: GraphQLFieldDefinition): DataFetcher<Cypher>?
+
+    fun input(name: String, type: GraphQLType): GraphQLArgument {
```

Comment:
is this the right place for this function? we have similar things in other places.